pkgs/development/tools/rust/cargo-leptos/default.nix

```diff
@@ -0,0 +1,31 @@
+{ stdenv, rustPlatform , fetchFromGitHub, Security, lib, pkg-config, openssl }:
+rustPlatform.buildRustPackage rec {
+  pname = "cargo-leptos";
+  version = "0.1.8";
+  buildFeatures = [ "no_downloads" ]; # cargo-leptos will try to download Ruby and other things without this feature
+
+  src = fetchFromGitHub {
+    owner = "leptos-rs";
+    repo = pname;
+    rev = version;
+    hash = "sha256-z4AqxvKu9E8GGMj6jNUAAWeqoE/j+6NoAEZWeNZ+1BA=";
+  };
+
+  cargoSha256 = "sha256-w/9W4DXbh4G5DZ8IGUz4nN3LEjHhL7HgybHqODMFzHw=";
+
+  nativeBuildInputs = [ pkg-config openssl ];
```

Comment:
Do these packages need to be listed as both `nativeBuildInputs` and `buildInputs`? Usually it's one or the other.

Reply:
`openssl` should be in only `buildInputs` as it's used by `openssl-sys`, and `pkg-config` only in `nativeBuildInputs` as it's required for cargo to find C deps. Not sure about Darwin stuff but I have no way to test